### PR TITLE
Fixe(Stock entry from donation) solve receipt rendering problem

### LINKED ISSUE #4751: Stock entry from donation throw an error during receipt generation
The stock entry from donation throw an error during receipt generation : 

image

### PR DESCRIPTION
**Pull request by jeremielodi**
Closes #4751

### CODE REVIEW

File: server/controllers/stock/reports/stock/entry_donation_receipt.js [hunk 1]
```diff
@@ -67,12 +67,13 @@ async function stockEntryDonationReceipt(documentUuid, session, options) {
...
   };
 
+  data.rows = rows;
+
   // sum elements of rows by their `total` property
   data.total = data.rows.reduce((aggregate, row) => {
     return row.total + aggregate;
   }, 0);
 
-  data.rows = rows;
```
Comment (jniles) [inline server/controllers/stock/reports/stock/entry_donation_receipt.js:75]: Is there anyway to write a test for this to make sure it doesn't happen again?

**Comment by jeremielodi** [comment]
As we already have test for PDF, I think I can give it a try.

**Comment by jniles** [comment]
Cool.  👍 .  To make it effective, it should fail without this change and pass with this change.

**Review by mbayopanda — APPROVED** [review]
LGTM 👍

**Comment by jniles** [comment]
@jeremielodi can you do a rebase with master?

**Comment by jniles** [comment]
@jeremielodi are you planning on writing a test for this?

**Comment by jeremielodi** [comment]
I'm writing the test for this

**Comment by jmcameron** [comment]
I have not tested this one so I'm not going to review it, but I was very impressed with the amount of tests that have been added.  Good job @jeremielodi !

File: test/integration/stock/stock.js [hunk 2]
```diff
@@ -7,205 +7,174 @@ const shared = require('./shared');
...
 
   // create new stock lots
-  it('POST /stock/lots create a new stock lots entry', () => {
-    return agent.post('/stock/lots')
-      .send(shared.movementFirstLots)
-      .then((res) => {
-        helpers.api.created(res);
-      })
-      .catch(helpers.handler);
+  it('POST /stock/lots create a new stock lots entry', async () => {
```
Comment (jniles) [inline test/integration/stock/stock.js:9]: async/await 💪

**Comment by jeremielodi** [comment]
No, I have just updated some tests by using ```async```. I added just one test for the donation's receipt rendering.

File: test/integration/stock/stock.js [hunk 3]
```diff
@@ -7,205 +7,174 @@ const shared = require('./shared');
...
+    const res = await agent.post('/stock/lots')
+      .send(shared.movementFirstLots);
+    helpers.api.created(res);
```
Comment (mbayopanda) [inline test/integration/stock/stock.js:12]: It seems like errors are not handled, i am wondering just if they can be handled

**Review by mbayopanda — APPROVED** [review]
👍 good job @jeremielodi 

I have made a comment about errors in e2e which are not handled, but since the test pass without error this comment is not a blocking one

File: test/integration/stock/stock.js (hunk 3 above)
Comment (jeremielodi) [reply test/integration/stock/stock.js:12]: If an error occurs the status code won't be 201

```js
api.created = function created(res) {

  // make sure the response has correct HTTP headers
  expect(res).to.have.status(201);
  expect(res).to.be.json;

```

File: test/integration/stock/stock.js [hunk 4]
```diff
@@ -7,205 +7,174 @@ const shared = require('./shared');
...
-        expect(res.body).to.be.deep.equal(mvtsByDocument);
-      })
-      .catch(helpers.handler);
+  it('POST /stock/lots/movements distribute lots to patients from a depot', async () => {
+
+    const res1 = await agent.post('/stock/lots/movements').send(shared.movementOutPatient);
+    helpers.api.created(res1);
+    // get details of the movement
+    const docUuid = res1.body.uuid;
+    const res2 = await agent.get(`/stock/lots/movements?document_uuid=${docUuid}`);
```
Comment (jniles) [inline test/integration/stock/stock.js:22]: This is really clean.  Great work @jeremielodi

**Review by jniles — APPROVED** [review]
@jeremielodi This is great!

I've made a few changes to tighten up the tests.  Basically, I:

1. Changed the text "Stock Exit to Patient" to "Stock Exit to Donation"
2. Changed from using a PDF renderer (that would spin up chrome) to using the JSON renderer.  It should be much quicker.
3. Removed optional UUIDs.

This looks great to me now.

**Pull request merged**